### PR TITLE
Feature/#434 refactor addon ui to use fla ui

### PR DESCRIPTION
**Pull request by MarekKosinski**

### CODE REVIEW

File: G1ANT.Addon.UI/Panels/UIControlsPanel.cs [hunk 1]
```diff
@@ -53,28 +58,28 @@ private string GetTreeNodeName(AutomationElement element)
             if (element == null)
                 return "";
             string id = "";
-            if (string.IsNullOrWhiteSpace(element.Current.AutomationId) == false)
-                id = $" #{element.Current.AutomationId}";
-            return $"{CutControlType(element.Current.ControlType.ProgrammaticName)}{id} \"{element.Current.Name}\"";
+            if (string.IsNullOrWhiteSpace(element.Properties.AutomationId.ValueOrDefault) == false)
```
Comment (wojszub) [inline G1ANT.Addon.UI/Panels/UIControlsPanel.cs]: !string.is....

File: G1ANT.Addon.UI/Panels/UIControlsPanel.cs [hunk 2]
```diff
@@ -53,28 +58,28 @@ private string GetTreeNodeName(AutomationElement element)
...
                 return null;
             StringBuilder result = new StringBuilder();
-
-            if (!string.IsNullOrWhiteSpace(element.Current.AutomationId))
-                result.AppendLine($"id: {element.Current.AutomationId}");
-            if (element.Current.ControlType != null)
+            
+            if (!string.IsNullOrWhiteSpace(element.Properties.AutomationId.ValueOrDefault))
+                result.AppendLine($"id: {element.Properties.AutomationId.ValueOrDefault}");
+            if (element.ControlType != null)
```
Comment (wojszub) [inline G1ANT.Addon.UI/Panels/UIControlsPanel.cs]: ControlType is enum, always != null

File: G1ANT.Addon.UI/Panels/UIControlsPanel.cs [hunk 3]
```diff
@@ -86,7 +91,8 @@ private void controlsTree_BeforeExpand(object sender, TreeViewCancelEventArgs e)
             {
                 if (e.Node.Tag is AutomationElement element)
                 {
-                    AutomationElement elem = TreeWalker.ControlViewWalker.GetFirstChild(element);
+                    var treeWalker = element.Automation.TreeWalkerFactory.GetControlViewWalker();
+                    AutomationElement elem = treeWalker.GetFirstChild(element);
```
Comment (wojszub) [inline G1ANT.Addon.UI/Panels/UIControlsPanel.cs]: var

File: G1ANT.Addon.UI/packages.config [hunk 4]
```diff
@@ -1,6 +1,8 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
   <package id="BouncyCastle" version="1.8.4" targetFramework="net452" />
+  <package id="FlaUI.Core" version="2.0.0-rc1" targetFramework="net452" />
+  <package id="FlaUI.UIA3" version="2.0.0-rc1" targetFramework="net452" />
```
Comment (wojszub) [inline G1ANT.Addon.UI/packages.config]: it's Release Candidate version

File: G1ANT.Addon.UI/Api/UIElement.cs [hunk 5]
```diff
@@ -74,94 +77,101 @@ public WPathStructure ToWPath(UIElement root = null)
...
             {
                 (invokePattern as InvokePattern)?.Invoke();
             }
-            else if (automationElement.TryGetCurrentPattern(SelectionItemPattern.Pattern, out var selectionPattern))
+            else if (automationElement.FrameworkAutomationElement.TryGetNativePattern<object>(SelectionItemPattern.Pattern, out var selectionPattern))
             {
                 (selectionPattern as SelectionItemPattern)?.Select();
             }
-            else if (automationElement.TryGetClickablePoint(out var pt))
+            if (automationElement.TryGetClickablePoint(out var pt))
```
Comment (pwitusik) [inline G1ANT.Addon.UI/Api/UIElement.cs]: missing "else if"

**Pull request merged**